graphene-writer/src/main/kotlin/com/graphene/writer/store/key/SimpleKeyStoreHandler.kt

```diff
@@ -31,133 +14,25 @@ import javax.annotation.PostConstruct
 @Component
 @ConditionalOnProperty(prefix = "graphene.writer.store.key.handlers.simple-key-store-handler", name = ["enabled"], havingValue = "true")
 class SimpleKeyStoreHandler(
-  private val elasticsearchFactory: ElasticsearchFactory,
-  private val property: SimpleKeyStoreHandlerProperty
-) : KeyStoreHandler, Runnable {
-
-  private val logger = Logger.getLogger(SimpleKeyStoreHandler::class.java)
-
-  private lateinit var client: TransportClient
-  private lateinit var scheduler: ScheduledExecutorService
-  private lateinit var request: MetricMultiGetRequestBuilder
-  private lateinit var index: String
-  private lateinit var type: String
-  private lateinit var bulkProcessor: BulkProcessor
-  private lateinit var grapheneKeyMapper: GrapheneKeyMapper
-
-  private var lastFlushTimeSeconds = currentTimeSeconds()
-  private var batchSize: Int = 0
-  private var flushInterval: Long = 0
-
-  private val metrics = ConcurrentLinkedQueue<GrapheneMetric>()
-
-  @PostConstruct
-  fun init() {
-    client = elasticsearchFactory.transportClient()
-
-    bulkProcessor = elasticsearchFactory.bulkProcessor(client)
-    grapheneKeyMapper = GrapheneKeyMapper()
-
-    index = property.index!!
-    type = property.type!!
-    request = MetricMultiGetRequestBuilder(client, index, type)
-    batchSize = property.bulk!!.actions
-    flushInterval = property.bulk!!.interval
-
-    scheduler = Executors.newSingleThreadScheduledExecutor(NamedThreadFactory("grapheneIndexThread"))
-    scheduler.scheduleWithFixedDelay(this, 3000, 100, TimeUnit.MILLISECONDS)
-  }
-
-  override fun handle(grapheneMetric: GrapheneMetric) {
-    metrics.offer(grapheneMetric)
+  elasticsearchFactory: ElasticsearchFactory,
+  property: SimpleKeyStoreHandlerProperty
+) : AbstractElasticsearchKeyStoreHandler(elasticsearchFactory, property) {
+
+  override fun source(tenant: String, graphiteKeyPart: String, depth: Int, leaf: Boolean): XContentBuilder {
+    return XContentFactory.jsonBuilder()
+      .startObject()
+      .field(TENANT, tenant)
+      .field(PATH, graphiteKeyPart)
+      .field(DEPTH, depth + 1)
+      .field(LEAF, leaf)
+      .endObject()
   }
 
-  override fun run() {
-    try {
-      val metric = metrics.poll()
-      if (Objects.nonNull(metric)) {
-        addToBatch(metric)
-      }
-    } catch (e: Exception) {
-      logger.error("Encountered error in busy loop: ", e)
-    }
-
-    if (0 < request.size()) {
-      flush()
-    }
-  }
-
-  private fun addToBatch(metric: GrapheneMetric) {
-    request.add(metric)
-
-    if (batchSize <= request.size() || lastFlushTimeSeconds < currentTimeSeconds() - flushInterval) {
-      flush()
-    }
+  companion object {
+    const val TENANT = "tenant"
+    const val DEPTH = "tenant"
```

Comment:
Seems DEPTH = "tenant" has to be changed if not intended.

Reply:
Please check this commit log:
https://github.com/graphene-monitoring/graphene/pull/7/commits/0ba7b5d524fbbf02ec8ef6d68a1e826539d47ff9